selector/random.go

```diff
@@ -0,0 +1,44 @@
+package selector
```

Comment:
why is this in package `selector` and not `selector/random`?

Reply:
I didn't realise this was possible, e.g. the encoding/json package is just named json: https://github.com/golang/go/blob/9326d71874730901c159643703a16dc34646e5bd/src/encoding/json/encode.go#L11

Reply:
sorry, was unclear in my comment. Should read "why is the implementation of random in package `selector` when it should be in package `random`?" But I guess it's explained per your comment about Default* impls